core/ts/__tests__/MaciState.test.ts

```diff
@@ -744,4 +720,416 @@ describe("MaciState", function () {
       });
     });
   });
+
+  describe("processMessage", () => {
+    const maciState = new MaciState(STATE_TREE_DEPTH);
+    const pollId = maciState.deployPoll(
+      BigInt(Math.floor(Date.now() / 1000) + duration),
+      maxValues,
+      treeDepths,
+      messageBatchSize,
+      coordinatorKeypair,
+    );
+
+    const poll = maciState.polls[pollId];
+
+    const user1Keypair = new Keypair();
+    // signup the user
+    const user1StateIndex = maciState.signUp(
+      user1Keypair.pubKey,
+      voiceCreditBalance,
+      BigInt(Math.floor(Date.now() / 1000)),
```

Comment:
because MaciState is used to represent the state of the smart contracts, and in order to re-create the same exact signup data, we need to be passing it as a parameter from the smart contracts event logs. Here in the unit test it doesn't really matter what we pass as we are not replicating the smart contract state, but you get the point